sys/mips/mips/elf_machdep.c

```diff
@@ -51,39 +51,59 @@ __FBSDID("$FreeBSD$");
 #include <machine/md_var.h>
 #include <machine/cache.h>
 
-#ifdef __mips_n64
-#ifndef COMPAT_CHERIABI
-struct sysentvec elf64_freebsd_sysvec = {
+#if __has_feature(capabilities)
+struct sysentvec elf64c_freebsd_sysvec = {
 	.sv_size	= SYS_MAXSYSCALL,
 	.sv_table	= sysent,
 	.sv_errsize	= 0,
 	.sv_errtbl	= NULL,
 	.sv_transtrap	= NULL,
+/* XXX: TODO */
+#if 1
 	.sv_fixup	= __elfN(freebsd_fixup),
 	.sv_sendsig	= sendsig,
 	.sv_sigcode	= sigcode,
 	.sv_szsigcode	= &szsigcode,
-	.sv_name	= "FreeBSD ELF64",
+#else
+	.sv_fixup	= cheriabi_elf_fixup,
+	.sv_sendsig	= cheriabi_sendsig,
+	.sv_sigcode	= cheri_sigcode,
+	.sv_szsigcode	= &szcheri_sigcode,
+#endif
+	.sv_name	= "CheriABI ELF64",
 	.sv_coredump	= __elfN(coredump),
 	.sv_imgact_try	= NULL,
-	.sv_minsigstksz	= MINSIGSTKSZ,
-	.sv_minuser	= VM_MIN_ADDRESS,
+	.sv_minsigstksz	= MINSIGSTKSZ,	/* XXXBD: or something bigger? */
+	.sv_minuser	= PAGE_SIZE,	/* Disallow mapping at NULL */
 	.sv_maxuser	= VM_MAXUSER_ADDRESS,
 	.sv_usrstack	= USRSTACK,
 	.sv_psstrings	= PS_STRINGS,
-	.sv_stackprot	= VM_PROT_ALL,
+	.sv_stackprot	= VM_PROT_READ | VM_PROT_WRITE,
+/* XXX: TODO */
+#if 1
 	.sv_copyout_strings = exec_copyout_strings,
 	.sv_setregs	= exec_setregs,
+#else
+	.sv_copyout_strings = cheriabi_copyout_strings,
+	.sv_setregs	= cheriabi_exec_setregs,
+#endif
 	.sv_fixlimit	= NULL,
 	.sv_maxssiz	= NULL,
-	.sv_flags	= SV_ABI_FREEBSD | SV_LP64 | SV_ASLR |
+	/* SV_ASLR? */
```

Comment:
I don't think we want ASLR for CheriABI.

Reply:
Ok, I can take it out in a future change.

Reply:
Went ahead and axed it from the original commit.